### PR TITLE
Move files between assetstores.

### PR DESCRIPTION
**Pull request by manthey**
Move files between assetstores.

Allow uploads to target a specific assetstore.

Fixed a bug in GridFS assetstore where, if enough data is in the assetstore, downloads could fail because there was no index just on n.

This still needs tests.

This resolves issue #330.

### CODE REVIEW

File: girder/utility/gridfs_assetstore_adapter.py [hunk 1]
```diff
@@ -65,6 +65,7 @@ def validateInfo(doc):
                 ('uuid', pymongo.ASCENDING),
                 ('n', pymongo.ASCENDING)
             ], unique=True)
+            chunkColl.create_index([('n', pymongo.ASCENDING)])
```
Comment (zachmullen) [inline girder/utility/gridfs_assetstore_adapter.py]: Why is this index required? Don't we always want to always filter by uuid?

Comment (manthey) [inline girder/utility/gridfs_assetstore_adapter.py]: In using Mongo version 3.2.9, I was getting an error when trying to download a large file that the data could not fit in memory and that an index was needed.  The lines
```
 cursor = self.chunkColl.find({
            'uuid': file['chunkUuid'],
            'n': {'$gte': n}
        }, projection=['data']).sort('n', pymongo.ASCENDING)
```
were not using the compound index.  Adding this index solves that.

Comment (zachmullen) [inline girder/utility/gridfs_assetstore_adapter.py]: Hm, that's odd. Any idea why that query wouldn't use the compound index? I would expect it to do so...

Comment (manthey) [inline girder/utility/gridfs_assetstore_adapter.py]: I don't know.  I will look into it further.

Comment (manthey) [inline girder/utility/gridfs_assetstore_adapter.py]: It turns out that somehow I had dropped the indices from the GridFS database.  I then uploaded a large file to it, and it wouldn't download due to the lack of any index on `n`.  The addition of the `n`-only index was a red herring.  We really just need to ensure the index exists when we attach to the database (in the `__init__` method), though we should still do so in the `validate` method as well.

**Comment by manthey** [comment]
This is ready for review.

File: girder/utility/gridfs_assetstore_adapter.py [hunk 2]
```diff
@@ -85,6 +85,10 @@ def __init__(self, assetstore):
                 self.assetstore.get('mongohost', None),
                 self.assetstore.get('replicaset', None)
             )[self.assetstore['db']].chunk
+            self.chunkColl.create_index([
+                ('uuid', pymongo.ASCENDING),
+                ('n', pymongo.ASCENDING)
+            ], unique=True)
```
Comment (zachmullen) [inline girder/utility/gridfs_assetstore_adapter.py]: If we want to call this logic in two places, let's make a small helper method.

**Comment by zachmullen** [comment]
Could you merge master into this for CI

**Comment by zachmullen** [comment]
LGTM

**Pull request merged**